addons/pos_loyalty/static/src/js/ProductScreen.js

```diff
@@ -56,7 +56,22 @@ export const PosLoyaltyProductScreen = (ProductScreen) =>
                     return false;
                 }
                 const trimmedCode = code.trim();
-                if (trimmedCode && trimmedCode.startsWith('044')) {
+                let nomenclatureRules = this.env.barcode_reader.barcode_parser.nomenclature.rules;
+                if (this.env.barcode_reader.fallbackBarcodeParser) {
+                    nomenclatureRules.push(...this.env.barcode_reader.fallbackBarcodeParser.nomenclature.rules);
+                }
+                const couponNomenclatureRules = _.filter(nomenclatureRules, function(rule) {
+                        return rule.type == "coupon";
+                    }
+                );
```

Comment:
Thanks, just one last thing:
```suggestion
                    return rule.type == "coupon";
                });
```